utilities/pipelines/resourceDeployment/New-TemplateDeployment.ps1

```diff
@@ -102,16 +102,21 @@ Optional. Maximum retry limit if the deployment fails. Default is 3.
 Optional. Do not throw an exception if it failed. Still returns the error message though
 
 .EXAMPLE
-New-DeploymentWithParameterFile -templateFilePath 'C:/KeyVault/deploy.json' -parameterFilePath 'C:/KeyVault/.test/parameters.json' -location 'WestEurope' -resourceGroupName 'aLegendaryRg'
+Invoke-InnerDeployment -templateFilePath 'C:/KeyVault/deploy.json' -parameterFilePath 'C:/KeyVault/.test/parameters.json' -location 'WestEurope' -resourceGroupName 'aLegendaryRg'
```

Comment:
Not sure about this one. Can it just be `New-Deployment` for simplicity?

Reply:
That was my first thought to. The only reason I didn't go with it was, that I suspected 
- a higher chance for conflicts with other cmdlets
- unclear responsibilities if we ever move those functions into dedicated files in a PowerShell module (like #2181)

Happy to change it, but just wanted to lay out my reasoning ;)

Reply:
What about `New-TemplateDeploymentInner` as we do in other scripts?

Reply:
Would be fine too.

Reply:
done

Reply:
It's now a hybrid `Invoke-TemplateDeploymentInner`. Not having a strong opinion besides that `New-TemplateDeploymentInner` would be more consistent with other scripts

Reply:
True, that was unintended. Adjusted